File: String-Algorithms/String-Algorithms-master/Permutation Checking/isPermutation.py
class solution:
    def isPermutation(self,input1,input2):
        mapp1 = []
        mapp2 = []
        for i in input1:
            if i not in mapp1:
                mapp1.append(i)
        for j in input2:
            if j not in mapp2:
                mapp2.append(j)
        mapp1.sort()
        mapp2.sort()
        return mapp1==mapp2

input1 = str(input())
input2 = str(input())
sol = solution()
print(sol.isPermutation(input1,input2)))

File: String-Algorithms/String-Algorithms-master/Permutation Checking/isPermutation.cpp
# include<iostream>
# include<cstdlib>
# include<string>
# include<map>

using namespace std;

class solution
{
public:
    bool isPermutation(string input1,string input2)
    {
        if(input1.size() != input2.size())
        {
            return false;
        }
        map<char,int> mapp1;
        map<char,int> mapp2;
        for(int i=0;i<input1.size();i++)
        {
            char key = input1[i];
            if(mapp1.count(key) == 0)
            {
                mapp1[key] = 1;
            }
            else
            {
                mapp1[key]++;
            }
        }
        for(int j=0;j<input2.size();j++)
        {
            char key = input2[j];
            if(mapp2.count(key) == 0)
            {
                mapp2[key] = 1;
            }
            else
            {
                mapp2[key]++;
            }
        }
        map<char,int>::iterator it = mapp1.begin();
        while(it != mapp1.end())
        {
            if(mapp1[it->first] != mapp2[it->first])
            {
                return false;
            }
            it++;
        }
        return true;
    }
};

int main()
{
    string input1,input2;
    getline(cin,input1);
    getline(cin,input2);
    solution sol;
    cout << sol.isPermutation(input1,input2)<< endl;
    return 0;
}
